### PR TITLE
Lab 6.1

### PR DESCRIPTION
**Pull request by miz14**

### CODE REVIEW

**Comment by Raipo7** [comment]
## [Codecov](https://app.codecov.io/gh/Raipo7/TeamN2/pull/26?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None) Report
> :exclamation: No coverage uploaded for pull request base (`main@33c1741`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None#section-missing-base-commit).
> Patch has no changes to coverable lines.

:exclamation: Your organization is not using the GitHub App Integration. As a result you may experience degraded service beginning May 15th. Please [install the Github App Integration](https://github.com/apps/codecov) for your organization. [Read more](https://about.codecov.io/blog/codecov-is-updating-its-github-integration/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

<details><summary>Additional details and impacted files</summary>


```diff
@@           Coverage Diff            @@
##             main       #26   +/-   ##
========================================
  Coverage        ?   100.00%           
========================================
  Files           ?        15           
  Lines           ?       433           
  Branches        ?        36           
========================================
  Hits            ?       433           
  Misses          ?         0           
  Partials        ?         0           
```




</details>

[:umbrella: View full report in Codecov by Sentry](https://app.codecov.io/gh/Raipo7/TeamN2/pull/26?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).   
:loudspeaker: Do you have feedback about the report comment? [Let us know in this issue](https://about.codecov.io/codecov-pr-comment-feedback/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=None).

File: SpaceBattle.Lib/ServerThread.cs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            while(!stop) {
+                strategy();
+            }
+        });
+    }
+    public void Start() {
+        thread.Start();
+    }
+
+    public void Stop() {
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/ServerThread.cs:39]: internal

File: SpaceBattle.Lib/ServerThread.cs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        strategy = newBehaviour;
+    }
+
+    internal void HandleCommand() {
+        var cmd = queue.Receive();
+        try {
+            cmd.Execute();
+        }
+        catch(Exception err) {
+            var exceptinHandlerStrategy = (IStrategy) new ExceptionFindHandlerStrategy().Execute(cmd, err);
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/ServerThread.cs:60]: IoC

File: SpaceBattle.Lib/ServerThread.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+public class SenderAdapter : ISender {
+    public BlockingCollection<ICommand> queue;
+
+    public SenderAdapter(BlockingCollection<ICommand> queue) {
+        this.queue = queue;
+    }
+
+    public void Send(object message) {
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/ServerThread.cs:183]: не object, а ICommand

File: SpaceBattle.Lib/ServerThread.cs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    }
+}
+
+public class SendCommand : ICommand {
+
+    ISender sender;
+
+    IEnumerable<ICommand> commandsList;
+
+    public SendCommand(string threadId, IEnumerable<ICommand> commandsList) {
```
Comment (zaryanezrya) [inline SpaceBattle.Lib/ServerThread.cs:194]: -IEnumerable

**Review by zaryanezrya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**